### DISCUSSION TITLE
Use switch  in fork task

### DISCUSSION THREAD

**Discussion by octoshikari** [Netflix/conductor#2754] [Q&A] [answered]
I see multiple execution for switch task when i use them inside FORK task.
Also same behaviour with DO_WHILE inside switch - for first iteration it execute task from parallel fork line

![image](https://user-images.githubusercontent.com/11928813/152802480-aec2a54a-46f2-4c09-b6ca-592f7b6f70e1.png)

**Comment by aravindanr** [comment, marked as answer]
hello @octoalex could you post the relevant workflow definition json and the execution json?

**Reply by octoshikari** [reply]
https://gist.github.com/octoalex/875f628a645620be85148fa280843748 - Execution

https://gist.github.com/octoalex/7b326883ccc247329c5da9713e075099 - definition

**Reply by apanicker-nflx** [reply]
Could you also provide some details about your deployment 
- conductor version
- type of persistence
- locking implementation

**Reply by octoshikari** [reply]
Sure.
Version 3.5.0
Postgres persistence.
Don't use locking

**Reply by apanicker-nflx** [reply]
This could be because of parallel threads trying to evaluate the same workflow simultaneously. We highly recommend using a workflow execution lock to prevent this race condition which causes undesirable side-effects on a workflow execution.
